packages-exp/messaging-exp/src/api/isSupported.ts

```diff
@@ -15,20 +15,7 @@
  * limitations under the License.
  */
 
-export function isSupported(): boolean {
-  if (self && 'ServiceWorkerGlobalScope' in self) {
-    // Running in ServiceWorker context
-    return isSWControllerSupported();
-  } else {
-    // Assume we are in the window context.
-    return isWindowControllerSupported();
-  }
-}
-
-/**
- * Checks to see if the required APIs exist.
- */
-function isWindowControllerSupported(): boolean {
+export async function isWindowSupported(): Promise<boolean> {
```

Comment:
Why is it async? It seems all checks are sync.

Reply:
ATM yes.
It's make to async firstly to make the other `isSupported` style. but more importmently to be able to check for idb.open